src/ast.ts

```diff
@@ -775,6 +776,14 @@ export abstract class Node {
     return false;
   }
 
+  /** Tests if this node is a literal of a string kind. */
+  get isStringLiteral(): bool {
+    if (this.kind == NodeKind.LITERAL) {
+      return (<LiteralExpression>changetype<Node>(this)).literalKind == LiteralKind.STRING;
+    }
+    return false;
```

Comment:
```suggestion
    return this.kind == NodeKind.LITERAL
        && (<LiteralExpression>changetype<Node>(this)).literalKind == LiteralKind.STRING;
```
I also remember that in another place, instead of resorting to `changetype`, the TS problem can also be solved with a `var self: Node = this`, which doesn't produce the TS error when cast. Not sure what's better, though.